zfs.go

```diff
@@ -224,24 +235,60 @@ func (d *Dataset) Mount(overlay bool, options []string) (*Dataset, error) {
 // ReceiveSnapshot receives a ZFS stream from the input io.Reader, creates a
 // new snapshot with the specified name, and streams the input data into the
 // newly-created snapshot.
-func ReceiveSnapshot(input io.Reader, name string) (*Dataset, error) {
+// It includes the option "-F" like boolean value.
+func ReceiveSnapshot(input io.Reader, name string, overwrite bool) (*Dataset, error) {
 	c := command{Command: "zfs", Stdin: input}
-	_, err := c.Run("receive", name)
-	if err != nil {
-		return nil, err
+	if overwrite == false {
+		_, err := c.Run("receive", name)
+		if err != nil {
+			return nil, err
+		}
+		return GetDataset(name)
+	} else {
+		_, err := c.Run("receive", "-F",name)
+		if err != nil {
+			return nil, err
+		}
+		return GetDataset(name)
 	}
-	return GetDataset(name)
 }
 
 // SendSnapshot sends a ZFS stream of a snapshot to the input io.Writer.
 // An error will be returned if the input dataset is not of snapshot type.
-func (d *Dataset) SendSnapshot(output io.Writer) error {
+// It includes the option "-R".
+func (d *Dataset) SendSnapshot(output io.Writer, flags SendFlag) error {
```

Comment:
I suppose it's not a super improvement. Really, I have done this changes to adapt it to my necessities. I can do more progress but more later. If you want to add more functionalities, go! Because I'm rookie in golang and I take a lot of time^^